salt/docker/init.sls

```diff
@@ -83,13 +83,18 @@ docker:
     - onlyif: systemctl status docker.service
     - onchanges:
       - /etc/systemd/system/docker.service.d/proxy.conf
-      - /etc/docker/daemon.json
     - require:
       - file: /etc/sysconfig/docker
-      - file: /etc/docker/daemon.json
```

Comment:
I would take this block out and create a `docker-restart-config` state, very similar to the `docker-reload-config`:

```SaltStack
docker-restart-config:
   cmd.run:
     - name: systemctl restart docker.service
     - onlyif: systemctl status docker.service
     - onchanges:
       - /etc/systemd/system/docker.service.d/proxy.conf
```

Reply:
I'd have to test it, but we can use `service.running` I guess:

```SaltStack
docker:
  service.running:
    - onchanges:
      - /etc/systemd/system/docker.service.d/proxy.conf
```

This should issue a `restart` when `/etc/systemd/system/docker.service.d/proxy.conf` changes.

Also,

```SaltStack
docker:
  service.running:
    - reload: True
    - onchanges:
      - /etc/systemd/system/docker.service.d/proxy.conf
```

This should issue a `reload` when `/etc/systemd/system/docker.service.d/proxy.conf` changes.

Reply:
Reference: https://github.com/saltstack/salt/blob/2016.11/salt/states/service.py#L596-L615

Reply:
I think that to have `service.running`, `reload` and `onchanges` in the same state is rather cryptic...

Reply:
I didn't mean to have them on the same state but rather use `service.running` than direct `systemctl` commands.